site/_posts/2022-11-01-authenticators-ceremonies-webauthn-oh-my.md

```diff
@@ -0,0 +1,128 @@
+
+{% include _image.liquid src="/assets/img/blogs/webauthn-terminology/registration-flow.png" alt="WebAuth Registration Flow diagram" class="img-fluid" figure=true %}
+
+1. The user visits the web application of the RP and initiates registration.
+2. The server creates a challenge key for registration, which is only used once.
+3. The challenge key, along with the user's info, is sent to the user's device.
+4. The user is prompted to authenticate themselves (this could be via biometrics, facial recognition system, or other forms depending on the device).
+5. After authentication success, the authenticator (the user's device) creates a new key pair (the public and private keys). 
+6. The authenticator uses the private key to sign the challenge and then sends the public key and a credential identifier alongside the challenge over to the server.
+7. The server validates the credibility of the private key by using the public key to evaluate the signed challenge.
+8. The server associates the credentials with the user's account for future authentication purposes and redirects them to the application.
+
+### The Authentication Flow
+
+After a user's device is registered with the RP, the user can be authenticated. There are six steps involved in the WebAuthn authentication process:
+
+{% include _image.liquid src="/assets/img/blogs/webauthn-terminology/authentication-flow.png" alt="WebAuth Authentication Flow diagram" class="img-fluid" figure=true %}
+
+The following are the steps for this authentication process:
+
+1. The user initiates login via the browser.
+2. The server (RP) creates and sends a unique challenge to the authenticator.
+3. The authenticator receives the challenge.
+4. The authenticator prompts the user for device authorization (again, via biometrics, facial recognition system, etc.).
+5. Upon successful authorization, the authenticator creates a cryptographic signature and sends it back to the server.
+6. The server verifies the signature of the unique challenge that it sent to the user (in step 2). Upon successful verification, the user is logged into the web application.
+
+## WebAuthn vs. FIDO 2
+
+WebAuthn was developed by the [World Wide Web Consortium (W3C)](http://www.w3.org/), which is the international standards organization for the world wide web. Meanwhile, Fast Identity Online 2 (FIDO2) was created by [FIDO Alliance](https://fidoalliance.org/) with the aim of creating passwordless authentication. WebAuthn is a core component of [FIDO2 specifications](https://fidoalliance.org/fido2/).
+
+The FIDO Alliance decided to partner with W3C to make FIDO2 authentication a standard for the web. WebAuthn became a part of the FIDO2 specification, and a [set of browser APIs](https://developer.mozilla.org/en-US/docs/Web/API/Web_Authentication_API) was finalized, which is widely used today for passwordless authentication.
```

Comment:
```suggestion
The FIDO Alliance decided to partner with W3C to make FIDO2 authentication a standard for the web. WebAuthn became a part of the FIDO2 specification, and a [set of browser APIs](https://developer.mozilla.org/en-US/docs/Web/API/Web_Authentication_API) was finalized, which are widely used today for passwordless authentication.
```

I guess "is" could be referring to the "set" rather than "APIs," but this sounds more correct.

These APIs are still being developed and expanded, but the core WebAuthn APIs used for registration and authentication are probably set.